src/auth.ts

```diff
@@ -0,0 +1,43 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+export const M2_DIR = '.m2';
+export const SETTINGS_FILE = 'settings.xml';
+
+export async function configAuthentication(username: string, password: string) {
+  if (username && password) {
+    core.debug(`configAuthentication with ${username} and a password`);
+    const directory: string = path.join(os.homedir(), M2_DIR);
+    await io.mkdirP(directory);
+    core.debug(`created directory ${directory}`);
```

Comment:
`core.debug` is fine. By we don't necessarily want users to have to enable step debugging to see anything. Certain things can be outputted normally.

Reply:
Ok, I'll adjust some of the output to include `console.log` output where it would be more generally useful.